Automattic/wp-calypso
# SitesList: convert onUpdatedPlugin method to a Redux action

This patch removes the `SitesList.onUpdatedPlugin` method and moves its functionality to Redux. The equivalent action to update the `state.sites.updates` state wasn't implemented until now.

What it does: when a plugin is manually updated to a new version, it decrements the "available updates" counters in `state.sites.updates`. The only consumer of that information is the `SiteIndicator` component.

There are two parts (commits):
- add `SITE_PLUGIN_UPDATED` action to Redux.
- use it instead of `SitesList.onUpdatedPlugin` and remove the old code.

**How to test:**
1. Download an older version of some plugin from WordPress.org (for example, I downloaded Akismet 4.0 from https://wordpress.org/plugins/akismet/advanced/)
2. Install it to a Jetpack site using the "Plugin Upload" feature (must be a "real" Jetpack site, not AT)
3. Reload Calypso and check that it shows an orange "updates available" indicator for the site.
4. Go to `/plugins/manage` and update the outdated plugin.
5. Verify that the orange site indicator disappears immediately after the update is finished.

The site indicator is connected to the `state.sites.updates` data, so if it changes correctly, it's a proof that the patch works as expected.

Here is a video of how the expected behavior looks like:
![plugin-update-screencast](https://user-images.githubusercontent.com/664258/33207973-09154112-d110-11e7-9152-8661055c7b3f.gif)
